arch/risc-v/src/esp32c3/esp32c3_i2c.c

```diff
@@ -1454,6 +1472,10 @@ static inline void esp32c3_i2c_process(struct esp32c3_i2c_priv_s *priv,
               if (priv->bytes == msg->length)
                 {
                   priv->i2cstate = I2CSTATE_STOP;
+                  if ((msg->flags & I2C_M_NOSTOP) != 0)
+                    {
+                      priv->i2cstate = I2CSTATE_FINISH;
+                    }
```

Comment:
```suggestion
                  if ((msg->flags & I2C_M_NOSTOP) == 0)
                    {
                      priv->i2cstate = I2CSTATE_STOP;
                    }
#ifndef CONFIG_I2C_POLLED
                  else
                    {
                      priv->i2cstate = I2CSTATE_FINISH;
                    }
#endif
```
`I2CSTATE_FINISH` does not apply for Polled I2C.